tests/conftest.py

```diff
@@ -127,14 +127,27 @@ def homedir(i18n):
 
 
 @pytest.fixture(scope="function")
-def functional_test_app_started_context(homedir, reply_status_codes, session, config, qtbot):
+def export_service():
+    """An export service that assumes the Qubes RPC calls are successful and skips them."""
+    export_service = export.Service()
+    # Ensure the export_service doesn't rely on Qubes OS:
+    export_service._run_disk_test = lambda dir: print("_run_disk_test")
+    export_service._run_usb_test = lambda dir: print("_run_usb_test")
+    export_service._run_disk_export = lambda dir, path, pasphrase: print("_run_disk_export")
+    return export_service
```

Comment:
This `export_service` was modified for testing purposes.

No matter how hacky disabling the Qubes OS bits is at the moment, that modification is nicely contained in a single place, and no other code needs to know about the details. For the rest of the tests, this is an `export.Service` instance like any other.

Note that this is made possible by a neat encapsulation of the Qubes OS-dependent commands in private methods :green_heart:

Reply:
Yep, I'm with you that for now this is the way to go and also am a fan of the qubes-specific bits being encapsulated. Makes it much easier to reason about and test the code. Thanks for putting so much work into thinking this through.